### ISSUE TITLE
gem dependency mismatch about thrift_client and thirft

### ISSUE THREAD

**Issue by kkismd** [cassandra-rb/cassandra#84]
in cassandra-0.10.0/lib/cassandra.rb 

```
gem 'thrift', '~> 0.5.0'
require 'thrift'
gem 'thrift_client', '~> 0.6.0'
require 'thrift_client'
```

it causes gem dependency error in below situation.

Gemfile:

```
source "http://rubygems.org"
gem "cassandra"
```

Gemfile.lock:

```
GEM
  remote: http://rubygems.org/
  specs:
    cassandra (0.10.0)
      json
      rake
      simple_uuid (>= 0.1.0)
      thrift_client (>= 0.6.0)
    json (1.5.1)
    rake (0.8.7)
    simple_uuid (0.1.2)
    thrift (0.6.0)
    thrift_client (0.6.2)
      thrift (~> 0.6.0)
```

thrift_client-0.6.2 requires thrift-0.6.0 or higher.

**Comment by rwjblue** [comment]
This is due to a change in thrift_client 0.6.2 (pushed 2011/05/09) which changes its own internal thrift dependency to ~> 0.6.0.

I have removed the requirement of thrift altogether as we should be letting thrift_client do it's own requirements and pushed that into master with the following commit:

https://github.com/fauna/cassandra/commit/3de72f0aa984dbcfa7190c287872c95c9497d60b

We need to validate that change, and push out a new gem version (probably to 0.11.0).

**Comment by ryanking** [comment]
I would push an 0.11, but there are failing tests (with ruby 1.8.7) Can you look into those?

**Comment by rwjblue** [comment]
I have been primarily testing in 1.9.2.  I'll take a look at it under 1.8.7, and try to get things cleaned up.

**Comment by rwjblue** [comment]
This is actually an issue with the underlying thrift update to 0.6.0. 

I am getting inconsistent errors while running the test suite.  The same test will pass once then error with a CassandraThrift::Cassandra::Client::TransportException when run a second time.

If we require thrift_client = 0.6.0 it will fix this dependency resolution issue, but long term we need to figure out what is going on in thrift-0.6.0 that is causing the underlaying issue.

**Comment by kkismd** [comment]
Thanks.

I understand that the problem is complicated.
At this time, I will take some workaround on our project.
something like...

Gemfile:

```
source "http://rubygems.org"
gem "cassandra", "0.10.0"
gem "thrift_client", "0.6.0"
```

Of course, I hope that the underlying issue is solved fundamentally.

**Comment by rwjblue** [comment]
The requirement issues are fixed with 0.11.0.  The test suite errors are related to the connect_timeout thrift option, and setting them slightly higher seems to have all tests in the green again.
